spec.emu

```diff
@@ -76,36 +145,49 @@ contributors: Ujjwal Sharma, Younies Mahmoud
-              1. Append the new Record { [[Type]]: _p_, [[Value]]: _num_ } to the end of _result_.
-        1. Else,
-          1. For each row in <emu-xref href="#table-duration-components"></emu-xref>, except the header row, in table order, do
-            1. Let _field_ be the Property value.
-            1. Let _val_ be _duration_.[[&lt;_field_&gt;]].
-            1. If _val_ is not *undefined* and _field_ exists in _durationFormat_.[[Fields]], then
-              1. Let _num_ be ! PartitionNumberPattern(_durationFormat_.[[NumberFormat]], _val_).
-              1. Let _pr_ be ! ResolvePlural(_durationFormat_.[[PluralRules]], _val_).
-              1. Let _dataLocale_ be _durationFormat_.[[DataLocale]].
-              1. Let _localeData_ be %DurationFormat%.[[LocaleData]].
-              1. Let _dataLocaleData_ be _localeData_.[[&lt;_dataLocale_&gt;]].
-              1. Let _template_ be _dataLocaleData_.[[formats]].[[&lt;_style_&gt;]].[[&lt;_field_&gt;]].[[&lt;_pr_&gt;]].
-              1. Let _parts_ be ! MakePartsList(_template_, _field_, _num_).
-              1. Let _concat_ be an empty String.
-              1. For each element _part_ in _parts_, in List order, do
-                1. Set _concat_ to the string-concatenation of _concat_ and _part_.[[Value]].
-                1. If _part_ has a [[Unit]] field, then
-                  1. Set _concat_ to the string-concatenation of _concat_ and _part_.[[Unit]].
-              1. Append the new Record { [[Type]]: _field_, [[Value]]: _concat_ } to the end of _result_.
-          1. Set _result_ to ! CreatePartsFromList(_durationFormat_.[[ListFormat]], _result_).
+        1. For each row in <emu-xref href="#table-duration-components"></emu-xref>, except the header row, in table order, do
+          1. Let _styleSlot_ be the Style Slot value.
+          1. Let _displaySlot_ be the Display Slot value.
+          1. Let _valueSlot_ be the Value Slot value.
+          1. Let _unit_ be the Unit value.
+          1. Let _style_ be _durationFormat_.[[&lt;_styleSlot_&gt;]].
+          1. Let _nfOpts_ be ! OrdinaryObjectCreate(`null`).
+          1. Let _value_ be `0`.
+          1. If _style_ is `"2-digit"` or `"numeric"` and _unit_ is `"seconds"` or below, then
+              1. Let _clipped_ be ! ClipDuration(_duration_, _unit_).
+              1. Let _totalOpts_ be ! ObjectCreate(`null`).
+              1. Perform ! CreateDataPropertyOrThrow(_totalOpts_, `"unit"`, _unit_).
+              1. Set _value_ to ? Call(_clipped_, `"total"`, _totalOpts_).
+              1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, `"maximumFractionDigits"`, _durationFormat_.[[FractionalDigits]]).
+          1. Else,
+            1. Set _value_ to _duration_.[[&lt;_valueSlot_&gt;]]
+          1. If _style_ is `"2-digit"`, then
+            1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, `"minimumIntegerDigits"`, `2`).
+          1. If _value_ is `0` or `-0` and _display_ is `"auto"`, then
```

Comment:
What about the case where `minutesStyle` is `2-digit` and `secondsDisplay` is `auto`?  I don't think you want to skip zeroes there.